internal/tools/document-lint/mdparse/diff/diff.go

```diff
@@ -0,0 +1,396 @@
+package diff
+
+import (
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/document-lint/mdparse/md"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/document-lint/mdparse/model"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/document-lint/mdparse/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/document-lint/mdparse/util"
+)
+
+// logic to load schema and markdown to print the diff
+
+// diff for resource first find the resource schema, then find the Markdown document file for it(or specify by string)
+
+type MissType int
```

Comment:
I would break this file up into check_timeouts.go check_reqopt.go check_default_value.go check_possible_values.go

where each file contains the "check type object" and the functions that do the checking, liking based on an interface so they can all be called with the same arguments